heron/ckptmgr/tests/java/com/twitter/heron/ckptmgr/CheckpointManagerServerTest.java

```diff
@@ -0,0 +1,387 @@
+  private static CheckpointManager.RegisterStMgrRequest registerStmgrRequest;
+  private static CheckpointManager.RegisterTMasterRequest registerTMasterRequest;
+
+  private static PhysicalPlans.Instance instance;
+
+  private CheckpointManagerServer checkpointManagerServer;
+  private SimpleCheckpointManagerClient simpleCheckpointManagerClient;
+
+  private NIOLooper serverLooper;
+  private NIOLooper clientLooper;
+  private ExecutorService threadPools;
+
+  private IStatefulStorage backendStorage;
+
+  @BeforeClass
+  public static void setup() throws Exception {
+    PhysicalPlans.InstanceInfo info = PhysicalPlans.InstanceInfo.newBuilder()
+        .setTaskId(TASK_ID)
+        .setComponentIndex(COMPONENT_INDEX)
+        .setComponentName(COMPONENT_NAME)
+        .build();
+
+    instance = PhysicalPlans.Instance.newBuilder()
+        .setInstanceId(INSTANCE_ID)
+        .setStmgrId(STMGR_ID)
+        .setInfo(info)
+        .build();
+
+    CheckpointManager.InstanceStateCheckpoint checkpoint =
+        CheckpointManager.InstanceStateCheckpoint.newBuilder()
+            .setCheckpointId(CHECKPOINT_ID)
+            .setState(ByteString.copyFrom(BYTES))
+            .build();
+
+    saveInstanceStateRequest = CheckpointManager.SaveInstanceStateRequest.newBuilder()
+        .setInstance(instance)
+        .setCheckpoint(checkpoint)
+        .build();
+
+    getInstanceStateRequest = CheckpointManager.GetInstanceStateRequest.newBuilder()
+        .setInstance(instance)
+        .setCheckpointId(CHECKPOINT_ID)
+        .build();
+
+    cleanStatefulCheckpointRequest = CheckpointManager.CleanStatefulCheckpointRequest.newBuilder()
+        .setCleanAllCheckpoints(true)
+        .setOldestCheckpointPreserved(CHECKPOINT_ID)
+        .build();
+
+    registerStmgrRequest = CheckpointManager.RegisterStMgrRequest.newBuilder()
+        .setTopologyId(TOPOLOGY_ID)
+        .setStmgrId(STMGR_ID)
+        .setTopologyName(TOPOLOGY_NAME)
+        .build();
+
+    registerTMasterRequest = CheckpointManager.RegisterTMasterRequest.newBuilder()
+        .setTopologyId(TOPOLOGY_ID)
+        .setTopologyName(TOPOLOGY_NAME)
+        .build();
+  }
+
+  @Before
+  public void before() throws Exception {
+    serverLooper = new NIOLooper();
+    clientLooper = new NIOLooper();
+    threadPools = Executors.newFixedThreadPool(2);
+
+    backendStorage = mock(IStatefulStorage.class);
+
+    HeronSocketOptions serverSocketOptions =
+        new HeronSocketOptions(100 * 1024 * 1024, 100,
+            100 * 1024 * 1024, 100,
+            5 * 1024 * 1024,
+            5 * 1024 * 1024);
+
+    serverPort = SysUtils.getFreePort();
+    checkpointManagerServer = new CheckpointManagerServer(
+        TOPOLOGY_NAME, TOPOLOGY_ID,
+        CHECKPOINT_MANAGER_ID, backendStorage,
+        serverLooper, SERVER_HOST, serverPort, serverSocketOptions);
+
+    runServer();
+    // let the server warm up
+    Thread.sleep(2 * 1000);
```

Comment:
Can we refactor how tests are run to not need this sleep (and all the others)? Tests that require sleep are likely going to be flaky. Also  Hardcoding sleep times in tests has slows down CI, which is already too long (this class spends 20 seconds just sleeping!).